libnavui-androidauto/build.gradle

```diff
@@ -38,6 +38,7 @@ dependencies {
     api(project(":libnavigation-android"))
     implementation(project(":libnavui-resources"))
     implementation(project(":libnavui-util"))
+    implementation(project(":libtrip-notification"))
```

Comment:
This is likely wrong as the notification module can be excluded from the build. But I have to re-use some of the resources from there. Is it OK to move them to `:libnavigation-base`?

Reply:
Yeah also some discussion on how exactly we will include these dependencies https://github.com/mapbox/mapbox-navigation-android/pull/5773#discussion_r865960135

Add this api, is an example of how we will put a strict requirement to make a new minimum version of the sdk. Because Interceptor became available in 2.5+

Reply:
If we hold off on this merge, and we publish a version of `ui-androidauto`. Technically, that version will be compatible with 2.4 and possibly lower versions

Reply:
Ok rebase, you'll find that we will need to bump the version of the sdk to a 2.5. It shouldn't need to reference the `:libtrip-notification`.

Reply:
I had to add dependency on `com.mapbox.navigation:notification`, which I think is wrong for the same reason (the dependency can be excluded from the build). 
@mapbox/navigation-android is it OK to move resources from `:libtrip-notification` to some shared module? Or is there a better solution?

Reply:
```
implementation("com.mapbox.navigation:notification:2.5.0-beta.3")
```

This seems like a good solution. Let's move forward with adding it here.

Seems like the [`libnavigation-android`](https://github.com/mapbox/mapbox-navigation-android/blob/362dc4703a2df6923674f6e4197940e7626374a3/libnavigation-android/build.gradle#L29-L36) would be another possible solution. But IMO, `androidauto` is not reason enough to add it there.

Reply:
> ```
> implementation("com.mapbox.navigation:notification:2.5.0-beta.3")
> ```
> 
> 
>      
>   
> 
> This seems like a good solution. Let's move forward with adding it here. Maybe it should be `api("com...`, except the wrapper interface provided here may suffice
> 
> 
The problem with this solution is that if someone excludes notification module like described in the [docs](https://docs.mapbox.com/android/navigation/guides/get-started/modularization/#tripnotification), then they will break android-auto functionality as it won't be able to find necessary resources.

Reply:
That's right. Nav SDK or Android Auto cannot have any strong references to implementation details of the default `TripNotification` module.

Looks like we have a duplication between `com.mapbox.navigation.ui.maneuver.model.TurnIconResources` and `com.mapbox.navigation.trip.notification.internal.NotificationTurnIconResources` which is documented in https://github.com/mapbox/mapbox-navigation-android/pull/4683#issuecomment-892245936 but I think it would make sense to move the `com.mapbox.navigation.trip.notification.internal.NotificationTurnIconResources` to `libnavigation-util` and keep that dependency as implementation detail of the `android-auto` module.

Reply:
I guess this means that I have to create a separate PR, which moves resources to another module, then wait for Nav SDK release with these changes, and only after that I can proceed with implementing the notification interceptor for Android Auto.

Reply:
Good idea to move resources, it seems like a good idea to also make the resources available through a public api. That may be out of scope of this chnage.

But I am confused why modularization is the blocker here. This will not work with modularization one way or another right? Developers will need to implement their own interceptor as well. That was the original approach

Reply:
👍  Good to move the resources though, that will be better. Also think these resources could be surfaced with an api that is not excluded https://github.com/mapbox/mapbox-navigation-android/pull/5803#discussion_r871677612

Reply:
Actually, making more sense to me now why it's blocking if we want to have this interceptor render the icons. Otherwise we would need to add a runtime check to see if the resources are available - best to just move resources.

Reply:
> which moves resources to another module, then wait for Nav SDK release with these changes,

Think we can update android-auto to build off snapshots while we're developing for the next versions 🤔